include/nbl/builtin/hlsl/emulated/float64_t.hlsl

```diff
@@ -0,0 +1,657 @@
+                }
+
+                if (emulated_float64_t_impl::isZero(lhsData))
+                    return bit_cast<this_t>(sign);
+
+                const uint64_t lhsRealMantissa = (ieee754::extractMantissa(lhsData) | (1ull << ieee754::traits<float64_t>::mantissaBitCnt));
+                const uint64_t rhsRealMantissa = ieee754::extractMantissa(rhsData) | (1ull << ieee754::traits<float64_t>::mantissaBitCnt);
+
+                int exp = lhsBiasedExp - rhsBiasedExp + int(ieee754::traits<float64_t>::exponentBias);
+
+                uint64_t2 lhsMantissaShifted = emulated_float64_t_impl::shiftMantissaLeftBy53(lhsRealMantissa);
+                uint64_t mantissa = emulated_float64_t_impl::divmod128by64(lhsMantissaShifted.x, lhsMantissaShifted.y, rhsRealMantissa);
+
+                const int msb = emulated_float64_t_impl::_findMSB(mantissa);
+                if(msb != -1)
+                {
+                    const int shiftAmount = 52 - msb;
+                    assert(shiftAmount >= 0);
+                    mantissa <<= shiftAmount;
+                    exp -= shiftAmount;
+                }
+
+                mantissa &= ieee754::traits<float64_t>::mantissaMask;
+
+                return bit_cast<this_t>(emulated_float64_t_impl::assembleFloat64(sign, uint64_t(exp) << ieee754::traits<float64_t>::mantissaBitCnt, mantissa));
+            }
+            else
+            {
+                //static_assert(false, "not implemented yet");
+                return bit_cast<this_t>(0xdeadbeefbadcaffeull);
+            }
+        }
+
+        emulated_float64_t operator/(const float rhs) NBL_CONST_MEMBER_FUNC
+        {
+            return bit_cast<this_t>(data) / create(rhs);
+        }
+
+        // relational operators
+        bool operator==(this_t rhs) NBL_CONST_MEMBER_FUNC
+        {
+            if (!FastMath)
+            {
+                if (tgmath::isNaN<uint64_t>(data) || tgmath::isNaN<uint64_t>(rhs.data))
+                    return false;
+                if (emulated_float64_t_impl::areBothZero(data, rhs.data))
+                    return true;
+            }
+
+            const emulated_float64_t xored = bit_cast<this_t>(data ^ rhs.data);
+            if ((xored.data & 0x7FFFFFFFFFFFFFFFull) == 0ull)
+                return true;
+
+            return !(xored.data);
+        }
+        bool operator!=(emulated_float64_t rhs) NBL_CONST_MEMBER_FUNC
+        {
+            if (!FastMath && (tgmath::isNaN<uint64_t>(data) || tgmath::isNaN<uint64_t>(rhs.data)))
+                return false;
+
+            return !(bit_cast<this_t>(data) == rhs);
+        }
+        bool operator<(emulated_float64_t rhs) NBL_CONST_MEMBER_FUNC
+        {
+            if (!FastMath)
+            {
+                if (tgmath::isNaN<uint64_t>(data) || tgmath::isNaN<uint64_t>(rhs.data))
+                    return false;
+                if (emulated_float64_t_impl::areBothInfinity(data, rhs.data))
+                    return false;
```

Comment:
shouldn't `-Inf < +Inf` return true and `+Inf < -Inf` return false?

i'd check they're both infinity and of the same sign